recipes/grpc/all/conanfile.py

```diff
@@ -0,0 +1,235 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class grpcConan(ConanFile):
+    name = "grpc"
+    description = "Google's RPC library and framework."
+    topics = ("conan", "grpc", "rpc")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/grpc/grpc"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
```

Comment:
No, `cmake_find_package` AND `cmake_find_package_multi` are required ;)

Reply:
but why `cmake_find_package` is required if we use `cmake_find_package_multi`? I think it supersedes

Reply:
I've edited my answer to give more explanation, it's very important to understand the difference.

Reply:
Yes, I wasn't aware there were also `find_package(... CONFIG)` lines. I would like not to mix them, but probably it is better to add both than to try to patch the project.

Reply:
Yes, patching upstream project would be odd, their CMakeLists is perfectly fine, it's how a good CMakeLists should be written.

Reply:
okay, so it seems like for generic case (e.g. template) it's safe to include both `cmake_find_package` + `cmake_find_package_multi`? especially, if I don't want to dig into that are built-in modules for my requires, and requires of requires, and so on.

Reply:
new `CMakeDeps` may utilize [CMAKE_FIND_PACKAGE_PREFER_CONFIG](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html#variable:CMAKE_FIND_PACKAGE_PREFER_CONFIG) in order to avoid exactly that situation

Reply:
It could be a solution, but it requires CMake >= 3.15. Why CMakeDeps doesn't generate both Find<pkg>.cmake and <pkg>-config.cmake files?

Reply:
> Why CMakeDeps doesn't generate both `Find<pkg>.cmake` and `<pkg>-config.cmake` files?

I guess because find modules are outdated and no longer recommended

Reply:
Nobody will right `find_package(ZLIB REQUIRED CONFIG)` in its CMakeLists, find module files are still needed, recommended or not.